cloud/aws/services/ec2/instances.go

```diff
@@ -199,10 +199,72 @@ func (s *Service) runInstance(i *v1alpha1.Instance) (*v1alpha1.Instance, error)
 		return nil, errors.Errorf("no instance returned for reservation %v", out.GoString())
 	}
 
-	return fromSDKTypeToInstance(out.Instances[0]), nil
+	return s.fromSDKTypeToInstance(out.Instances[0]), nil
 }
 
-func fromSDKTypeToInstance(v *ec2.Instance) *v1alpha1.Instance {
+// UpdateInstanceSecurityGroups modifies the security groups of the given
+// EC2 instance.
+func (s *Service) UpdateInstanceSecurityGroups(instanceID *string, securityGroups []*string) error {
+	input := &ec2.ModifyInstanceAttributeInput{
+		InstanceId: instanceID,
+		Groups:     securityGroups,
+	}
+
+	_, err := s.EC2.ModifyInstanceAttribute(input)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateResourceTags updates the tags for an instance.
+// This will be called if there is anything to create (update) or delete.
+// We may not always have to perform each action, so we check what we're
+// receiving to avoid calling AWS if we don't need to.
+func (s *Service) UpdateResourceTags(resourceID *string, create map[string]string, delete map[string]string) error {
+	// If we have anything to create or update
+	if len(create) > 0 {
+		// Convert our create map into an array of *ec2.Tag
+		createTagsInput := s.mapToTag(create)
+
+		// Create the CreateTags input.
+		input := &ec2.CreateTagsInput{
+			Resources: []*string{resourceID},
+			Tags:      createTagsInput,
+		}
+
+		// Create/Update tags in AWS.
+		_, err := s.EC2.CreateTags(input)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If we have anything to delete
+	if len(delete) > 0 {
+		// Convert our delete map into an array of *ec2.Tag
+		deleteTagsInput := s.mapToTag(delete)
+
+		// Create the DeleteTags input
+		input := &ec2.DeleteTagsInput{
+			Resources: []*string{resourceID},
+			Tags:      deleteTagsInput,
+		}
+
+		// Delete tags in AWS.
+		_, err := s.EC2.DeleteTags(input)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// fromSDKTypeToInstance takes a ec2.Instance and returns our v1.alpha1.Instance
+// type. EC2 types are wrapped or converted to our own types here.
+func (s *Service) fromSDKTypeToInstance(v *ec2.Instance) *v1alpha1.Instance {
```

Comment:
I moved this function to a method on `Service` since it needed to access some of the `Service` helper methods for dealing with `Tags` and `SecurityGroups`.

Reply:
Could we make `tagToMap` or `mapToTag` functions instead of methods?

Reply:
We could. I could leave them in `tags.go` since that seems to be a good logical place for them, move them out of the `Service` struct and move `fromSDKTypeToInstance` back out of `Service`.

Reply:
(If there is a better place that standalone functions like that belong, I could also move them there)

Reply:
Yeah I like that idea! For now it seems that `tags.go` is a good place to but those functions in. Later on, we can think about putting all these conversion functions in a different place for sure, we should probably open an issue to track that.

Reply:
Done, they're moved out of `Service` and are standalone functions now. `fromSDKTypeToInstance` has been restored to its former functionness.